src/main/java/org/beanpod/switchboard/repository/LogRepository.java

```diff
@@ -11,4 +12,10 @@
   LogEntity save(LogEntity logEntity);
 
   List<LogEntity> findAll();
+
+  // The Query covers three cases: 1- serialNumber = X,Y 2- serialNumber = Y,X 3- serialNumber = X
+  @Query(
+      "SELECT l FROM LogEntity l WHERE l.serialNumber LIKE CONCAT(?1,',%') "
+          + "OR l.serialNumber LIKE CONCAT('%,',?1) OR l.serialNumber = ?1")
+  List<LogEntity> findBySerialNumber(String serialNumber);
 }
```

Comment:
i think you can use the already defined like or contains method,
https://www.baeldung.com/spring-jpa-like-queries

Reply:
We got three cases for the serial number format, so I don't think we can match it using a single wildcard. Also, I haven't gone with the contain method since it could produce bugs. Let's say you want to match logs with serial number 1, then 13,4 is matched as well since it contains 1.

Reply:
that makes sense